File: leetcode/104.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        if(!(root)){
            //if self is nullptr, which is not node, but is endding sign
            return 0;
        }
        //make sure ererything you need can be control in this block
        int l_node=1+maxDepth((*root).left);//1 is self depth, maxDepth is child length
        int r_node=1+maxDepth((*root).right);

        return std::max(l_node,r_node);//max function is belong to Algorithm header file
    }
};

File: leetcode/104.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        big=0
        #is immutable object in parameter change?, no ,always different object if edit
        def recur(root,depth):
            nonlocal big
            if root==None:
                return 0
            if depth>big:
                big=depth
            if root.left:
                recur(root.left,depth+1)
            if root.right:
                recur(root.right,depth+1)
        recur(root,1)
        return big